Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -262,6 +262,50 @@ public class ChatChannelController: DataController, DelegateCallable, DataStoreP
         }
     }
 
+    /// Updates  channel information with provided data, and removes unneeded properties.
+    ///
+    /// - Parameters:
+    ///   - team: New team.
+    ///   - members: New members.
+    ///   - invites: New invites.
+    ///   - extraData: New `ExtraData`.
+    ///   - unsetProperties: Properties from the channel that are going to be cleared/unset.
+    ///   - completion: The completion. Will be called on a **callbackQueue** when the network request is finished.
+    ///                 If request fails, the completion will be called with an error.
+    ///
+    public func partialChannelUpdate(
+        name: String? = nil,
+        imageURL: URL? = nil,
+        team: String? = nil,
+        members: Set<UserId> = [],
+        invites: Set<UserId> = [],
+        extraData: [String: RawJSON] = [:],
+        unsetProperties: [String] = [],
```

Comment:
Which properties are valid to be unset? Can we improve the interface and expose an enum or a struct with static properties to make the API easier to use? Otherwise, customers must dig into the codebase or check the documentation to see which properties can be unset and how they are written.

Something as we have in `ChannelCapability` would be ideal since it is also `RawRepresentable`, so customers can also use a regular String.

Reply:
The API for other SDKs is String-based, and there is no documentation as to which properties are possible to be unset. We could be adding those but would make it pretty maintenance heavy. Not sure if it's worth it. `FilterKey` is a good example where, IMO, it makes more sense to have such approach.